Problem: hctl shutdown leaves hare-consul-agent.service in failed state

This leads to situation when cluster can't be bootstrapped again.

hare-consul-agent status:
```
ssc-vm-0581 # systemctl status hare-consul-agent
● hare-consul-agent.service - Consul agent for Hare
   Loaded: loaded (/usr/lib/systemd/system/hare-consul-agent.service; disabled; vendor preset: disabled)
   Active: failed (Result: exit-code) since Wed 2020-08-12 12:19:42 MDT; 40min ago
  Process: 3439 ExecStart=/opt/seagate/cortx/hare/libexec/consul/hare-consul (code=exited, status=1/FAILURE)
 Main PID: 3439 (code=exited, status=1/FAILURE)
   CGroup: /system.slice/hare-consul-agent.service
           ├─3796 bash /opt/seagate/cortx/hare/libexec/elect-rc-leader
           ├─3806 gawk { print strftime("%Y-%m-%d %H:%M:%S"), $0 }
           ├─3964 bash /opt/seagate/cortx/hare/libexec/elect-rc-leader
           └─3965 consul watch -type=keyprefix -prefix eq/ /opt/seagate/cortx/hare/libexec/proto-rc
```
`hctl shutdown` doesn't consider cluster as running (which is true in general):
```
ssc-vm-0581 # hctl shutdown
Cluster is not running
```
Logs:
```
Aug 12 12:17:05 ssc-vm-0581.colo.seagate.com hare-consul[3439]: 2020-08-12T12:17:05.161-0600 [WARN]  agent.server.raft: heartbeat timeout reached, starting election: last-leader=
Aug 12 12:17:05 ssc-vm-0581.colo.seagate.com hare-consul[3439]: 2020-08-12T12:17:05.162-0600 [INFO]  agent.server.raft: entering candidate state: node="Node at 192.168.0.7:8300 [Candidate]" term=2
Aug 12 12:17:05 ssc-vm-0581.colo.seagate.com hare-consul[3439]: 2020-08-12T12:17:05.166-0600 [INFO]  agent.server.raft: election won: tally=1
Aug 12 12:17:05 ssc-vm-0581.colo.seagate.com hare-consul[3439]: 2020-08-12T12:17:05.166-0600 [INFO]  agent.server.raft: entering leader state: leader="Node at 192.168.0.7:8300 [Leader]"
Aug 12 12:17:05 ssc-vm-0581.colo.seagate.com hare-consul[3439]: 2020-08-12T12:17:05.166-0600 [INFO]  agent.server: cluster leadership acquired
Aug 12 12:17:05 ssc-vm-0581.colo.seagate.com hare-consul[3439]: 2020-08-12T12:17:05.167-0600 [INFO]  agent.server: New leader elected: payload=localhost
Aug 12 12:17:05 ssc-vm-0581.colo.seagate.com hare-consul[3439]: Processing server acl mode for: localhost - 0
Aug 12 12:17:05 ssc-vm-0581.colo.seagate.com hare-consul[3439]: 2020-08-12T12:17:05.169-0600 [INFO]  agent.server: Cannot upgrade to new ACLs: leaderMode=0 mode=0 found=true leader=192.168.0.7:8300
Aug 12 12:17:05 ssc-vm-0581.colo.seagate.com hare-consul[3439]: 2020-08-12T12:17:05.172-0600 [INFO]  agent.leader: started routine: routine="CA root pruning"
Aug 12 12:17:05 ssc-vm-0581.colo.seagate.com hare-consul[3439]: 2020-08-12T12:17:05.172-0600 [INFO]  agent.server: member joined, marking health alive: member=localhost
Aug 12 12:17:06 ssc-vm-0581.colo.seagate.com hare-consul[3439]: 2020-08-12T12:17:06.850-0600 [INFO]  agent: Caught: signal=hangup
Aug 12 12:17:06 ssc-vm-0581.colo.seagate.com hare-consul[3439]: 2020-08-12T12:17:06.850-0600 [INFO]  agent: Reloading configuration...
Aug 12 12:17:06 ssc-vm-0581.colo.seagate.com hare-consul[3439]: BootstrapExpect is set to 1; this is the same as Bootstrap mode.
Aug 12 12:17:06 ssc-vm-0581.colo.seagate.com hare-consul[3439]: bootstrap = true: do not enable unless necessary
Aug 12 12:17:06 ssc-vm-0581.colo.seagate.com hare-consul[3439]: 2020-08-12T12:17:06.865-0600 [ERROR] agent: Failed to run watch handler: watch_handler=[/opt/seagate/cortx/hare/libexec/consul-watch-handler] error="exit status 1"
Aug 12 12:17:06 ssc-vm-0581.colo.seagate.com sudo[3781]:     root : TTY=unknown ; PWD=/ ; USER=root ; COMMAND=/bin/mkdir -p /var/log/hare
Aug 12 12:17:06 ssc-vm-0581.colo.seagate.com hare-consul[3439]: 2020-08-12T12:17:06.872-0600 [ERROR] agent: Failed to run watch handler: watch_handler=[/opt/seagate/cortx/hare/libexec/consul-watch-handler] error="exit status 1"
Aug 12 12:17:06 ssc-vm-0581.colo.seagate.com hare-consul[3439]: 2020-08-12T12:17:06.872-0600 [ERROR] agent: Failed to run watch handler: watch_handler=[/opt/seagate/cortx/hare/libexec/consul-watch-handler] error="exit status 1"
Aug 12 12:17:06 ssc-vm-0581.colo.seagate.com hare-consul[3439]: 2020-08-12T12:17:06.873-0600 [ERROR] agent: Failed to invoke http watch handler: watch=http://192.168.0.7:8008 error="Post http://192.168.0.7:8008: dial tcp 192.168.0.7:8008: connect: connection refused"
Aug 12 12:17:06 ssc-vm-0581.colo.seagate.com hare-consul[3439]: 2020-08-12T12:17:06.873-0600 [ERROR] agent: Failed to invoke http watch handler: watch=http://192.168.0.7:8008 error="Post http://192.168.0.7:8008: dial tcp 192.168.0.7:8008: connect: connection refused"
Aug 12 12:17:06 ssc-vm-0581.colo.seagate.com hare-consul[3439]: 2020-08-12T12:17:06.874-0600 [ERROR] agent: Failed to invoke http watch handler: watch=http://192.168.0.7:8008/watcher/bq error="Post http://192.168.0.7:8008/watcher/bq: dial tcp 192.168.0.7:8008: connect: connection refused"
Aug 12 12:17:06 ssc-vm-0581.colo.seagate.com hare-consul[3439]: 2020-08-12T12:17:06.874-0600 [ERROR] agent: Failed to invoke http watch handler: watch=http://192.168.0.7:8008 error="Post http://192.168.0.7:8008: dial tcp 192.168.0.7:8008: connect: connection refused"
Aug 12 12:17:07 ssc-vm-0581.colo.seagate.com hare-consul[3439]: 2020-08-12T12:17:07.822-0600 [INFO]  agent: Synced node info
Aug 12 12:17:07 ssc-vm-0581.colo.seagate.com hare-consul[3439]: 2020-08-12T12:17:07.825-0600 [INFO]  agent: Synced service: service=6
Aug 12 12:17:07 ssc-vm-0581.colo.seagate.com hare-consul[3439]: 2020-08-12T12:17:07.822-0600 [INFO]  agent: Synced node info
Aug 12 12:17:07 ssc-vm-0581.colo.seagate.com hare-consul[3439]: 2020-08-12T12:17:07.825-0600 [INFO]  agent: Synced service: service=6
Aug 12 12:17:07 ssc-vm-0581.colo.seagate.com hare-consul[3439]: 2020-08-12T12:17:07.827-0600 [INFO]  agent: Synced service: service=9
Aug 12 12:17:07 ssc-vm-0581.colo.seagate.com hare-consul[3439]: 2020-08-12T12:17:07.830-0600 [ERROR] agent: Failed to invoke http watch handler: watch=http://192.168.0.7:8008 error="Post http://192.168.0.7:8008: dial tcp 192.168.0.7:8008: connect: connection refused"
Aug 12 12:17:07 ssc-vm-0581.colo.seagate.com hare-consul[3439]: 2020-08-12T12:17:07.835-0600 [INFO]  agent: Synced service: service=12
Aug 12 12:17:07 ssc-vm-0581.colo.seagate.com hare-consul[3439]: 2020-08-12T12:17:07.851-0600 [ERROR] agent: Failed to invoke http watch handler: watch=http://192.168.0.7:8008 error="Post http://192.168.0.7:8008: dial tcp 192.168.0.7:8008: connect: connection refused"
Aug 12 12:17:10 ssc-vm-0581.colo.seagate.com sudo[3967]:     root : TTY=unknown ; PWD=/ ; USER=root ; COMMAND=/bin/mkdir -p /var/log/hare
Aug 12 12:17:13 ssc-vm-0581.colo.seagate.com hare-consul[3439]: 2020-08-12T12:17:13.045-0600 [INFO]  agent: Synced check: check=service:6
Aug 12 12:17:16 ssc-vm-0581.colo.seagate.com hare-consul[3439]: 2020-08-12T12:17:16.651-0600 [INFO]  agent: Synced check: check=service:9
Aug 12 12:17:17 ssc-vm-0581.colo.seagate.com hare-consul[3439]: 2020-08-12T12:17:17.658-0600 [INFO]  agent: Synced check: check=service:9
Aug 12 12:17:20 ssc-vm-0581.colo.seagate.com hare-consul[3439]: 2020-08-12T12:17:20.721-0600 [INFO]  agent: Synced check: check=service:9
Aug 12 12:17:23 ssc-vm-0581.colo.seagate.com hare-consul[3439]: 2020-08-12T12:17:23.101-0600 [ERROR] agent: Newer Consul version available: new_version=1.8.2 current_version=1.7.2
Aug 12 12:17:28 ssc-vm-0581.colo.seagate.com hare-consul[3439]: 2020-08-12T12:17:28.669-0600 [INFO]  agent: Synced check: check=service:12
Aug 12 12:17:29 ssc-vm-0581.colo.seagate.com hare-consul[3439]: 2020-08-12T12:17:29.689-0600 [INFO]  agent: Synced check: check=service:12
Aug 12 12:17:31 ssc-vm-0581.colo.seagate.com hare-consul[3439]: 2020-08-12T12:17:31.708-0600 [INFO]  agent: Synced check: check=service:12
Aug 12 12:18:03 ssc-vm-0581.colo.seagate.com hare-consul[3439]: 2020-08-12T12:18:03.319-0600 [INFO]  agent: Synced check: check=service:12
Aug 12 12:18:11 ssc-vm-0581.colo.seagate.com hare-consul[3439]: 2020-08-12T12:18:11.458-0600 [INFO]  agent: Synced check: check=service:9
Aug 12 12:18:11 ssc-vm-0581.colo.seagate.com hare-consul[3439]: 2020-08-12T12:18:11.460-0600 [ERROR] agent: Failed to invoke http watch handler: watch=http://192.168.0.7:8008 error="Post http://192.168.0.7:8008: dial tcp 192.168.0.7:8008: connect: connection refused"
Aug 12 12:19:41 ssc-vm-0581.colo.seagate.com systemd[1]: Stopping Consul agent for Hare...
Aug 12 12:19:41 ssc-vm-0581.colo.seagate.com hare-consul[3439]: 2020-08-12T12:19:41.058-0600 [INFO]  agent: Caught: signal=terminated
Aug 12 12:19:41 ssc-vm-0581.colo.seagate.com hare-consul[3439]: 2020-08-12T12:19:41.058-0600 [INFO]  agent: Graceful shutdown disabled. Exiting
Aug 12 12:19:41 ssc-vm-0581.colo.seagate.com hare-consul[3439]: 2020-08-12T12:19:41.058-0600 [INFO]  agent: Requesting shutdown
Aug 12 12:19:41 ssc-vm-0581.colo.seagate.com hare-consul[3439]: 2020-08-12T12:19:41.058-0600 [INFO]  agent.server: shutting down server
Aug 12 12:19:41 ssc-vm-0581.colo.seagate.com hare-consul[3439]: 2020-08-12T12:19:41.058-0600 [WARN]  agent.server.serf.lan: serf: Shutdown without a Leave
Aug 12 12:19:41 ssc-vm-0581.colo.seagate.com hare-consul[3439]: 2020-08-12T12:19:41.058-0600 [WARN]  agent.server.serf.wan: serf: Shutdown without a Leave
Aug 12 12:19:41 ssc-vm-0581.colo.seagate.com hare-consul[3439]: 2020-08-12T12:19:41.058-0600 [INFO]  agent.server.router.manager: shutting down
Aug 12 12:19:41 ssc-vm-0581.colo.seagate.com hare-consul[3439]: 2020-08-12T12:19:41.059-0600 [INFO]  agent: consul server down
Aug 12 12:19:41 ssc-vm-0581.colo.seagate.com hare-consul[3439]: 2020-08-12T12:19:41.059-0600 [INFO]  agent: shutdown complete
Aug 12 12:19:41 ssc-vm-0581.colo.seagate.com hare-consul[3439]: 2020-08-12T12:19:41.059-0600 [INFO]  agent: Stopping server: protocol=DNS address=127.0.0.1:8600 network=tcp
Aug 12 12:19:41 ssc-vm-0581.colo.seagate.com hare-consul[3439]: 2020-08-12T12:19:41.059-0600 [INFO]  agent: Stopping server: protocol=DNS address=192.168.0.7:8600 network=tcp
Aug 12 12:19:41 ssc-vm-0581.colo.seagate.com hare-consul[3439]: 2020-08-12T12:19:41.059-0600 [INFO]  agent: Stopping server: protocol=DNS address=127.0.0.1:8600 network=udp
Aug 12 12:19:41 ssc-vm-0581.colo.seagate.com hare-consul[3439]: 2020-08-12T12:19:41.059-0600 [INFO]  agent: Stopping server: protocol=DNS address=192.168.0.7:8600 network=udp
Aug 12 12:19:41 ssc-vm-0581.colo.seagate.com hare-consul[3439]: 2020-08-12T12:19:41.059-0600 [INFO]  agent: Stopping server: protocol=HTTP address=127.0.0.1:8500 network=tcp
Aug 12 12:19:42 ssc-vm-0581.colo.seagate.com hare-consul[3439]: 2020-08-12T12:19:42.059-0600 [WARN]  agent: Timeout stopping server: protocol=HTTP address=127.0.0.1:8500 network=tcp
Aug 12 12:19:42 ssc-vm-0581.colo.seagate.com hare-consul[3439]: 2020-08-12T12:19:42.059-0600 [INFO]  agent: Stopping server: protocol=HTTP address=192.168.0.7:8500 network=tcp
Aug 12 12:19:42 ssc-vm-0581.colo.seagate.com hare-consul[3439]: 2020-08-12T12:19:42.059-0600 [INFO]  agent: Waiting for endpoints to shut down
Aug 12 12:19:42 ssc-vm-0581.colo.seagate.com hare-consul[3439]: 2020-08-12T12:19:42.059-0600 [INFO]  agent: Endpoints down
Aug 12 12:19:42 ssc-vm-0581.colo.seagate.com hare-consul[3439]: 2020-08-12T12:19:42.059-0600 [INFO]  agent: Exit code: code=1
Aug 12 12:19:42 ssc-vm-0581.colo.seagate.com systemd[1]: hare-consul-agent.service: main process exited, code=exited, status=1/FAILURE
Aug 12 12:19:42 ssc-vm-0581.colo.seagate.com systemd[1]: Stopped Consul agent for Hare.
Aug 12 12:19:42 ssc-vm-0581.colo.seagate.com systemd[1]: Unit hare-consul-agent.service entered failed state.
Aug 12 12:19:42 ssc-vm-0581.colo.seagate.com systemd[1]: hare-consul-agent.service failed.
```

--- a/systemd/consul-server-conf.json.in
+++ b/systemd/consul-server-conf.json.in
@@ -1,5 +1,6 @@
 {
   "server": true,
+  "leave_on_terminate": true,
   "watches": [
     {
       "type": "key",

--- a/utils/hare-bootstrap
+++ b/utils/hare-bootstrap
@@ -122,10 +122,16 @@ abort_if_RC_leader_election_is_impossible() {
 
         for svc in hare-hax $confd_id; do
             if $ssh sudo systemctl --quiet --state=failed is-failed $svc; then
-                ok=false
-                status_commands+=(
-                    "${ssh:+$ssh }sudo systemctl status $svc"
-                )
+                # hare-hax service fails during stop which blocks the further
+                # deployment of the cluster. Remove this explicit reset-failed
+                # and enable code after this statement once the stop failure is
+                # resolved.
+                sudo systemctl reset-failed $svc
+
+                #ok=false
+                #status_commands+=(
+                #    "${ssh:+$ssh }sudo systemctl status $svc"
+                #)
             fi
         done
         if $ok; then

